modelon/impact/client/client.py

```diff
@@ -1,20 +1,99 @@
 """This module provides an entry-point to the client APIs."""
 import logging
+from typing import List, Optional
+from dataclasses import dataclass
 from semantic_version import SimpleSpec, Version  # type: ignore
 import modelon.impact.client.configuration
-from modelon.impact.client.entities.workspace import WorkspaceDefinition, Workspace
-from modelon.impact.client.entities.project import ProjectDefinition, Project
+from modelon.impact.client.entities.project import Project, ProjectDefinition, VcsUri
+from modelon.impact.client.entities.workspace import (
+    WorkspaceDefinition,
+    Workspace,
+    VcsReference,
+)
 import modelon.impact.client.sal.service
 import modelon.impact.client.sal.exceptions
 import modelon.impact.client.credential_manager
 import modelon.impact.client.jupyterhub
+from modelon.impact.client.operations.workspace import WorkspaceImportOperation
 from modelon.impact.client.sal.uri import URI
 from modelon.impact.client import exceptions
 
 
 logger = logging.getLogger(__name__)
 
 
+def _get_vcs_uri_from_reference_id(
+    shared_definition: WorkspaceDefinition, reference_id: str
+):
+    projects = shared_definition.projects + shared_definition.dependencies
+    for project in projects:
+        if (
+            isinstance(project.reference, VcsReference)
+            and project.reference.id == reference_id
+        ):
+            return project.reference.vcsUri
+    raise ValueError(
+        f"The {reference_id} doesn't exist in workspace projects or dependencies!"
+    )
+
+
+@dataclass
+class Selection:
+    entry_id: str
+    project: Project
+
+    def to_dict(self):
+        return {'id': self.entry_id, 'project': {'id': self.project.id}}
+
+
+@dataclass
+class ProjectMatching:
+    entry_id: str
+    vcs_uri: str
+    projects: List[Project]
+
+    def get_selection(self, index: int) -> Selection:
+        return Selection(entry_id=self.entry_id, project=self.projects[index])
+
+    def make_selection_interactive(self) -> Selection:
+        if len(self.projects) == 1:
+            print(f"Only one project matches the URI {self.vcs_uri}!")
+            selection = self.get_selection(index=0)
+            print(
+                f"Resolving conflict automatically using project with ID: "
+                f"{selection.project.id} for repository with URI: {self.vcs_uri}."
+            )
+            return selection
+
+        print(f"Multiple existing projects match the URI {self.vcs_uri}!")
+        for i, project in enumerate(self.projects):
+            print(f"{i+1}. {project.definition.name} {project.vcs_uri}")
```

Comment:
I wound if it makes sense to map between 0 and 1 index here. The user probably can handle it and we are in Python land after all.

Maybe we can look at how other tools for nice clear output? For example I like 'update-alternatives':
```
There are 3 choices for the alternative java (providing /usr/bin/java).

  Selection    Path                                      Priority   Status
------------------------------------------------------------
* 0            /usr/lib/jvm/java-6-openjdk/jre/bin/java   1061      auto mode
  1            /usr/bin/gij-4.4                           1044      manual mode
  2            /usr/lib/jvm/java-6-openjdk/jre/bin/java   1061      manual mode
  3            /usr/lib/jvm/java-6-sun/jre/bin/java       63        manual mode

Press enter to keep the current choice[*], or type selection number:
```
Which would look something like:
```
  Selection    Name                   URI
------------------------------------------------------------
  0            Awesome project        git.com
  1            So and so project      svn.com
```

It is not too hard to format things with columns in Python also: https://stackoverflow.com/a/9996049.

Of course this is just polish but would be nice for it to be flashy when you do a demo 😄